app/helpers.go

```diff
@@ -77,7 +86,60 @@ func GetTopics(ctx context.Context, username string, _topics []string) ([]string
 	return topics, err
 }
 
-func authenticate(ctx context.Context, app *App, userID string, topics ...string) (bool, []string, error) {
+// IsAuthorized returns a boolean indicating whether the user is authorized to read messages
+// from at least one of the given topics, and also a slice of all topics on which the user has authorization.
+func IsAuthorized(ctx context.Context, app *App, userID string, topics ...string) (bool, []string, error) {
+	httpAuthEnabled := app.Config.GetBool("httpAuth.enabled")
+
+	if httpAuthEnabled {
+		return httpAuthenticate(app, userID, topics)
+	}
+
+	return mongoAuthenticate(ctx, userID, topics)
+}
+
+func httpAuthenticate(app *App, userID string, topics []string) (bool, []string, error) {
+	timeout := app.Config.GetDuration("httpAuth.timeout") * time.Second
+	address := app.Config.GetString("httpAuth.requestURL")
+
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	isAuthorized := false
+	allowedTopics := make([]string, 0)
+	for _, topic := range topics {
+		authRequest := authRequest{
+			Username: userID,
+			Topic:    topic,
+		}
+
+		jsonPayload, _ := json.Marshal(authRequest)
+		request, _ := http.NewRequest(http.MethodPost, address, bytes.NewReader(jsonPayload))
+
+		credentialsNeeded := app.Config.GetBool("httpAuth.iam.enabled")
+		if credentialsNeeded {
+			username := app.Config.GetString("httpAuth.iam.credentials.username")
+			password := app.Config.GetString("httpAuth.iam.credentials.password")
+
+			request.SetBasicAuth(username, password)
+		}
+
+		response, err := client.Do(request)
+		if err != nil {
+			return false, nil, err
+		}
+
+		if response.StatusCode == 200 {
+			isAuthorized = true
+			allowedTopics = append(allowedTopics, topic)
+		}
+	}
+
+	return isAuthorized, allowedTopics, nil
+}
+
+func mongoAuthenticate(ctx context.Context, userID string, topics []string) (bool, []string, error) {
```

Comment:
nit: could you call this mongoAuthorize? Authentication != authorization

Reply:
Nice catch, done!